File: Data Structures & Algorithms/find-pivot-index/submission-3.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        total_sum = sum(nums)
        left_sum = 0
        
        for i, num in enumerate(nums):
            # Calculate the right sum on the fly
            right_sum = total_sum - left_sum - num
            #print(i, num, left_sum, right_sum)
            if left_sum == right_sum:
                return i
                
            left_sum += num
            
        return -1

File: Data Structures & Algorithms/sqrtx/submission-3.cpp
class Solution {
public:
    int mySqrt(int x) {
        int l = 0, r = x;
        int res = 0;
        while(l <= r) {
            int m = l + (r-l)/2;
            long long num = (long long)m*m;
            if(num == x) {
                return m;
            } else if( x < num) {
                r = m - 1;
            } else {
                res = m;
                l = m + 1;
            }
        }

        return res;
    }
};